pkg/proxy/winkernel/hcntesting/fakehcn.go

```diff
@@ -0,0 +1,275 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcntesting
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/Microsoft/hcsshim/hcn"
+	"k8s.io/klog/v2"
+)
+
+const (
+	guid                        = "123ABC"
+	destinationPrefix           = "192.168.2.0/24"
+	isolationId                 = 4096
+	providerAddress             = "10.0.0.3"
+	distributedRouterMacAddress = "00-11-22-33-44-55"
+	macRangeStartAddress        = "00-15-5D-52-C0-00"
+	macRangeEndAddress          = "00-15-5D-52-CF-FF"
+	ipAddressPrefix             = "192.168.1.0/24"
+	subnetNextHop               = "192.168.1.1"
+	subnetDestinationPrefix     = "0.0.0.0/0"
+)
+
+//the fakeHCN saves the created endpoints and loadbalancers in slices to be able to work with them easier
+type FakeHCN struct {
+	Endpoints     []*hcn.HostComputeEndpoint
+	Loadbalancers []*hcn.HostComputeLoadBalancer
+}
+
+func (HCN *FakeHCN) GetNetworkByName(networkName string) (*hcn.HostComputeNetwork, error) {
```

Comment:
These mock functions always give same hardcoded responses. Need to have variable results based on the input parameters.